src/testdrive/src/action/kafka/verify_data.rs

```diff
@@ -428,15 +432,23 @@ fn parse_expected_messages(
             },
         };
 
-        ensure!(
-            deserializer.next().is_none(),
-            "at most two records per expect line"
-        );
+        let content =
+            str::from_utf8(content).context("internal error: contents were previously a string")?;
+        let partition = match content.trim().split_once("=") {
+            None => None,
```

Comment:
it might still be worth keeping the `ensure` check around in this case to avoid confusing in a test case when there is more than one record accidentally added to a row